### DISCUSSION TITLE
Are Angular signals removed from the browser memory after the component is destroyed?

### DISCUSSION THREAD

**Discussion by 7KK2** [community/community#132527] [Programming Help]
### Body

0

I am using the new Angular feature named signals like this:

arr: WritableSignal<string[]> = signal([]);

I want to know if this signal is destroyed from browser memory after the component is destroyed, because this array can get quite big and maybe will be constructed and destroyed several times.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by nikolozkoiava** [comment]
If the signal is inside a component, then the contents of the component will get destroyed, so yes, the signal also is destroyed.

I think it will be destroyed but will stay in memory, until garbage collection runs in the browser to remove it permanently.

**Comment by ebndev** [comment]
Hi @7KK2 & @nikolozkoiava, thanks for being a part of the GitHub Community!

While we appreciate you wanting to contribute to Discussions, please note: We made the decision to disable the ability to earn Achievements in our Community in order to discourage users from participating in coordinated or inauthentic activity like rapid questions and answers in order to earn badges. You can learn more about this decision in our announcement post here [Achievements will no longer be available in the Community](https://github.com/orgs/community/discussions/106536).

Note that GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this, and other related posts. 

Any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
